### DISCUSSION TITLE
Track list vs. arranger canvas parts gradients

### DISCUSSION THREAD

**Discussion by terminator356** [muse-sequencer/muse#1205] [Team Posts]
Hi @kybos Been meaning to ask for a while, there's an obvious discrepancy
 between the track list and part gradients.
Could we add a wee bit of gradient to the track list?
Maybe in another theme, Dark Gradient?
Were you planning on making Dark Flat have flat parts as well?
That would make sense, then make another theme with gradients, I suppose.

The gradient is actually more functional than for looks.
A gradient sort of kind of allows that some similar foreground object colours
 will be seen at least a bit, more so when expanded vertically.

It would also enhance a selected track.
The large yellow swath makes it hard to see the yellow solo icons for example.

**Comment by kybos** [comment]
That's right, the look is currently somewhat incoherent in this respect.
I was already looking into making the parts flat as well, but I remember it was not as simple as just removing the gradients. The font in the part names didn't look good then etc., so I haven't followed on that yet. Partly because also I actually quite like the gradients in the parts.
I didn't like them in the track list, however (and in some other places), they were one of the elements that gave the old themes so dated look. But perhaps they could be made more decent for the new themes, maybe they will work better with the darker colors. 
This is surely one of the areas I am going to look into before the final release. I stopped making functional changes anyway and am trying to concentrate on optimizing the themes and other visual elements (also quite busy in the job ATM, so not much time left).

**Comment by kybos** [comment]
I tried out a couple of things.
The parts really don't look good when completely without gradients. But they look nice and fresh with decent gradients IMO, and the track list also gains some plasticity when given a slight gradient (but only slight, so it doesn't acquire that ugly protruding look).

![image](https://user-images.githubusercontent.com/10009541/110345095-800b3080-802e-11eb-8223-586be1127106.png)

**Comment by terminator356** [comment]
Decent. Maybe just a wee bit more.

You'll probably hate this idea, but it's an obvious one:
We have adjustable opacity. (I couldn't think of a sensible fixed value so I made it adjustable to user's taste.)
We have adjustable colours.
So how about an adjustable gradient? (Didn't I see something like that in Ardour?)
It would likely only apply to large areas like canvases and track list, because we have a few
 controls and effects rack and so on that use their own gradients. (Well, at least when no stylesheet
 is applied I think.)
And they are painted areas, not things that are controlled by stylesheet I think.
If you'd like, you don't have to apply the full adjustable gradient percentage to the track list,
 it could be more flat than the parts for example.
You could start with those two, and as we go along, other editors could be made to use it as well.

Just an idea. Although I guess it could no longer be called flat, eh? It defeats the purpose and name. ;-)

**Comment by kybos** [comment]
I had the same idea (as I see that the gradients are matter of taste to some extent).
So it's already implemented, I'm just testing and optimizing. 
So you can do things like this if you really wish (this is the maximum setting):

![image](https://user-images.githubusercontent.com/10009541/110916247-e3d87680-8318-11eb-80a0-2734f879f942.png)

The default for Dark Flat theme should be "almost flat", to keep up with the name, as the gradients can now easily be adjusted by the user (separately for tracks and parts, as the parts need more because of the transparency). 
I am reluctant to create new themes just because of the gradient, because I've learned that each theme means a lot of additional maintenance and testing for me, when I am adding/changing UI-related things. 
The only thing I've planned to add at some point is a new light theme in the style of Breeze, but it's a long-term plan, since a good theme needs a lot of time...

**Comment by kybos** [comment]
It's now in master.
The corresponding settings are under Appearance->Themes.

**Comment by terminator356** [comment]
Cool! You da man!
Yeah I forgot to suggest NOT allowing the gradient adjustment to go all the way to zero
 since that would look silly. So I think that's what you have done here. A reasonable range, I suppose.

The picture looks kind of... strange? Not sure why. Looks like candy?
Maybe gradient not starting at correct position?
Will pull this weekend and check 'er out.

Thanks bro.

**Comment by kybos** [comment]
As I wrote above, this picture is just a showcase for the maximum/extreme setting, certainly not something one would use in practice.
